### ISSUE TITLE
a issue about top_k_scores_test:  {'MAP': [0.0], 'NDCG': [0.0], 'jaccard': [0.0]}

### ISSUE THREAD

**Issue by twodog41** [YuxiaWu/RAG4DyG#2]
Dear author, I am a student who is trying to reproduce your project. After processing the dialog dataset, I ran the bash scripts/SimpleDyG/train_dialog.sh command to try to run the Evaluation_SimPleDyG.py file, but the terminal output is as shown in the title: top_k_scores_test: {'MAP': [0.0], 'NDCG': [0.0], 'jaccard': [0.0]}. No matter how many epochs are trained, these key indicators are always 0. After debugging, I found that the model has the following performance:
1. The model repeatedly generates the same token in the verification phase. The initial performance is: special tokens such as <|endofpre|> are repeatedly repeated in the generated sequence. In subsequent debugging, the model no longer outputs <|endofpre|>, but still repeatedly generates ordinary tokens (such as token ID 220), resulting in invalid and meaningless output.
2. Verification indicators (such as NDCG@1, NDCG@5) are always 0, which may also be that the model does not predict valid targets in vocab at all.
3. High-frequency tokens are mostly repeated items, and the top-5 logits outputs are almost the same at each step, indicating that the model has an obvious tendency to repeat patterns and it is difficult to jump out of the cycle.

I tried: adding code to successfully prohibit the model from generating special tokens, and trying to add sampling parameters, but neither completely solved the problem, and the values ​​of NDCG@1 and NDCG@5 are always 0.

I currently have difficulty finding the cause of these problems, so I humbly ask you for advice. Any form of help means a lot to me. Thank you for taking your time. The following are some related settings. I am not sure if these are useful. I hope this will help you judge. I only trained one epoch in this training to save time to observe the changes in NDCG@1 and NDCG@5:

vocab size:  50257
model.config GPT2Config {
  "_num_labels": 2,
  "activation_function": "gelu_new",
  "architectures": [
    "GPT2LMHeadModel"
  ],
  "attn_pdrop": 0.1,
  "bos_token_id": 50256,
  "do_sample": false,
  "early_stopping": false,
  "embd_pdrop": 0.1,
  "eos_token_ids": [
    50256
  ],
  "finetuning_task": null,
  "id2label": {
    "0": "LABEL_0",
    "1": "LABEL_1"
  },
  "initializer_range": 0.02,
  "is_decoder": false,
  "label2id": {
    "LABEL_0": 0,
    "LABEL_1": 1
  },
  "layer_norm_epsilon": 1e-05,
  "length_penalty": 1.0,
  "max_length": 20,
  "max_token_id": 50257,
  "model_type": "gpt2",
  "n_ctx": 1024,
  "n_embd": 256,
  "n_head": 2,
  "n_layer": 2,
  "n_positions": 1024,
  "num_beams": 1,
  "num_return_sequences": 1,
  "output_attentions": false,
  "output_hidden_states": false,
  "output_past": true,
  "pad_token_id": null,
  "pruned_heads": {},
  "repetition_penalty": 1.0,
  "resid_pdrop": 0.1,
  "summary_activation": null,
  "summary_first_dropout": 0.1,
  "summary_proj_to_labels": true,
  "summary_type": "cls_index",
  "summary_use_proj": true,
  "task_specific_params": {
    "text-generation": {
      "do_sample": true,
      "max_length": 50
    }
  },
  "temperature": 1.0,
  "top_k": 50,
  "top_p": 1.0,
  "torchscript": false,
  "use_bfloat16": false,
  "vocab_size": 50278
}

model GPT2LMHeadModel(
  (transformer): GPT2Model(
    (wte): Embedding(50278, 256)
    (wpe): Embedding(1024, 256)
    (drop): Dropout(p=0.1, inplace=False)
    (h): ModuleList(
      (0-1): 2 x Block(
        (ln_1): LayerNorm((256,), eps=1e-05, elementwise_affine=True)
        (attn): Attention(
          (c_attn): Conv1D()
          (c_proj): Conv1D()
          (attn_dropout): Dropout(p=0.1, inplace=False)
          (resid_dropout): Dropout(p=0.1, inplace=False)
        )
        (ln_2): LayerNorm((256,), eps=1e-05, elementwise_affine=True)
        (mlp): MLP(
          (c_fc): Conv1D()
          (c_proj): Conv1D()
          (act): NewGELUActivation()
          (dropout): Dropout(p=0.1, inplace=False)
        )
      )
    )
    (ln_f): LayerNorm((256,), eps=1e-05, elementwise_affine=True)
  )
  (lm_head): Linear(in_features=256, out_features=50278, bias=False)
)
Training/evaluation parameters Namespace(run_seed=True, n_gpu=1, timestamp='15', dataset='dialog', train_data_file='./resources/dialog/15/train.link_prediction', output_dir='simpledyg_ckpt/dialog/15/{7}/gpt2', model_type='gpt2', eval_data_file='./resources/dialog/15/val.link_prediction', eval_data_gt_file='./resources/dialog/15/val_gt.link_prediction', test_data_file='./resources/dialog/15/test.link_prediction', test_data_gt_file='./resources/dialog/15/test_gt.link_prediction', n_layer=2, n_head=2, n_embed=256, node_feat_file=None, should_continue=False, model_name_or_path='/home/wpk/RAG4DyG-main/gpt2_local', config_name=None, tokenizer_name=None, cache_dir=None, block_size=512, do_train=True, do_eval=False, evaluate_during_training=False, per_gpu_train_batch_size=32, per_gpu_eval_batch_size=32, gradient_accumulation_steps=1, learning_rate=0.0001, weight_decay=0.0, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=1.0, max_steps=-1, warmup_steps=0, logging_steps=500, save_steps=250, save_total_limit=None, eval_all_checkpoints=True, no_cuda=False, overwrite_cache=False, seed=7, fp16=False, fp16_opt_level='O1', local_rank=-1, patience=10, device=device(type='cuda'), para_names=['dataset', 'method', 'time', 'nlayer', 'nhead', 'nemb', 'bz', 'lr', 'seed'], para_values=['dialog', 'SimpleDyG', '15', 2, 2, 256, 32, 0.0001, 7], run_name='dataset_dialog_method_SimpleDyG_time_15_nlayer_2_nhead_2_nemb_256_bz_32_lr_0.0001_seed_7_', spl_tokens=['<|history|>', '<|endofhistory|>', '<|pre|>', '<|endofpre|>', '<|time0|>', '<|time1|>', '<|time2|>', '<|time3|>', '<|time4|>', '<|time5|>', '<|time6|>', '<|time7|>', '<|time8|>', '<|time9|>', '<|time10|>', '<|time11|>', '<|time12|>', '<|time13|>', '<|time14|>', '<|time15|>'])
file_path ./resources/dialog/15/train.link_prediction
***** Running training *****
  Num examples = 7464
  Num Epochs = 1.0
  Instantaneous batch size per GPU = 32
  Total train batch size (w. parallel, distributed & accumulation) = 32
  Gradient Accumulation steps = 1
  Total optimization steps = 233.0
  Starting fine-tuning.
Iteration: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 233/233 [00:43<00:00,  5.34it/s]
1591it [05:34,  4.76it/s]█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎| 232/233 [00:43<00:00,  4.35it/s]
Epoch: 0 | Step: 233 | train loss: 4.7501937780257455  | val_NDCG@5: 0.0 | lr: 0.0 
  val_NDCG@1 = 0.0
Epoch: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [06:18<00:00, 378.10s/it]
***** Train cost time: 0.10502703871991899 hours *****
***** Running val *****
1591it [05:46,  4.60it/s]
top_k_scores_val:  {'MAP': [0.0], 'NDCG': [0.0], 'jaccard': [0.0]}
***** Running test *****
1591it [04:29,  5.90it/s]
top_k_scores_test:  {'MAP': [0.0], 'NDCG': [0.0], 'jaccard': [0.0]}
 global_step = 233, average loss = 4.7501937780257455
